core/fallbackautoresolve6188_test.go

```
@@ -0,0 +1,143 @@
+  "created": 1700000000,
+  "model": "gpt-4o-mini",
+  "choices": [{"index": 0, "message": {"role": "assistant", "content": "ok"}, "finish_reason": "stop"}],
+  "usage": {"prompt_tokens": 5, "completion_tokens": 2, "total_tokens": 7}
+}`
+
+func TestFallbackRunsWhenProviderCannotBeAutoResolved(t *testing.T) {
+	var fallbackHits atomic.Int32
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		fallbackHits.Add(1)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprint(w, autoResolveChatBody)
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprint` is not checked

(errcheck)